### ISSUE TITLE
Issue in running the prediction

### ISSUE THREAD

**Issue by crypt3dC4rb0n** [deepdoctection/deepdoctection#12]
**Bug** 💥
1. Issue 1 - Error in running `df = analyzer.analyze(path=path)` . After debugging, I found that this was due to the wrong argument order passed to `DetectionResult` in `deepdoctection/extern/d2/d2.py`
```
Traceback (most recent call last):
  File "/home/xyz/repos/deepdoctection/run.py", line 14, in <module>
    page = next(doc)
  File "/home/xyz/repos/deepdoctection/venv/lib/python3.9/site-packages/dataflow/dataflow/common.py", line 315, in __iter__
    for dp in self.ds:
  File "/home/xyz/repos/deepdoctection/venv/lib/python3.9/site-packages/dataflow/dataflow/common.py", line 315, in __iter__
    for dp in self.ds:
  File "/home/xyz/repos/deepdoctection/venv/lib/python3.9/site-packages/dataflow/dataflow/common.py", line 315, in __iter__
    for dp in self.ds:
  [Previous line repeated 5 more times]
  File "/home/xyz/repos/deepdoctection/venv/lib/python3.9/site-packages/dataflow/dataflow/common.py", line 316, in __iter__
    ret = self.func(copy(dp))  # shallow copy the list
  File "/home/xyz/repos/deepdoctection/deepdoctection/pipe/base.py", line 81, in pass_datapoint
    self.serve(dp)
  File "/home/xyz/repos/deepdoctection/deepdoctection/pipe/layout.py", line 61, in serve
    detect_result_list = self.predictor.predict(dp.image)  # type: ignore
  File "/home/xyz/repos/deepdoctection/deepdoctection/extern/d2detect.py", line 137, in predict
    return self._map_category_names(detection_results)
  File "/home/xyz/repos/deepdoctection/deepdoctection/extern/d2detect.py", line 149, in _map_category_names
    result.class_name = self._categories_d2[str(result.class_id)]
KeyError: '0.9865013957023621'
```
Fix used - 
In `deepdoctection/extern/d2/d2.py`, changed
```
results = [
        DetectionResult(
            instances[k].pred_boxes.tensor.tolist()[0],
            instances[k].scores.tolist()[0],
            instances[k].pred_classes.tolist()[0],
        )
        for k in range(len(instances))
    ]
```
to 
```
results = [
        DetectionResult(
            instances[k].pred_boxes.tensor.tolist()[0],
            instances[k].pred_classes.tolist()[0],
            instances[k].scores.tolist()[0],
        )
        for k in range(len(instances))
    ]
```
2. Issue 2- After fixing the above error, I ran into another issue while iterating over the predictions, `page = next(doc)`
```
Traceback (most recent call last):
  File "/home/xyz/repos/deepdoctection/run.py", line 16, in <module>
    for page in df:
  File "/home/xyz/repos/deepdoctection/venv/lib/python3.9/site-packages/dataflow/dataflow/common.py", line 315, in __iter__
    for dp in self.ds:
  File "/home/xyz/repos/deepdoctection/venv/lib/python3.9/site-packages/dataflow/dataflow/common.py", line 315, in __iter__
    for dp in self.ds:
  File "/home/xyz/repos/deepdoctection/venv/lib/python3.9/site-packages/dataflow/dataflow/common.py", line 315, in __iter__
    for dp in self.ds:
  File "/home/xyz/repos/deepdoctection/venv/lib/python3.9/site-packages/dataflow/dataflow/common.py", line 316, in __iter__
    ret = self.func(copy(dp))  # shallow copy the list
  File "/home/xyz/repos/deepdoctection/deepdoctection/pipe/base.py", line 81, in pass_datapoint
    self.serve(dp)
  File "/home/xyz/repos/deepdoctection/deepdoctection/pipe/text.py", line 83, in serve
    detect_result_list = self.predictor.predict(predictor_input)  # type: ignore
  File "/home/xyz/repos/deepdoctection/deepdoctection/extern/tessocr.py", line 78, in predict
    text_lines=self.config.LINES,
  File "/home/xyz/repos/deepdoctection/deepdoctection/utils/metacfg.py", line 44, in __getattr__
    raise AttributeError(name)
AttributeError: LINES

```

**Additional context** 🧬
File `"/home/xyz/repos/deepdoctection/run.py"` is a script I created referring the `Get_Started.ipynb` for debugging the issue.

```
from matplotlib import pyplot as plt
from deepdoctection.analyzer import get_dd_analyzer

analyzer = get_dd_analyzer(language='deu')

path = "notebooks/pics/samples/sample_2"
df = analyzer.analyze(path=path)

doc=iter(df)
page = next(doc)

print(page.height, page.width, page.file_name)
image = page.viz()

plt.figure(figsize = (25,17))
plt.axis('off')
plt.imshow(image)

print(len(page.tables))
print(page.tables[0])

page.save("res.json")

```

And btw, Thanks for sharing your work. I was working on a similar implementation and found your repo insightful.

**Comment by JaMe76** [comment]
Hello and thank you for pointing to these issues and for providing your error analysis !

1.) Has been fixed with 7e9c07f .

2.) Apologize for that, this is a consequence from my part for not implementing enhancements thoroughly and merging them 
too quickly into the main branch. I will do my best to avoid these issues in the future.

The AttributeError arises from an old config file in the cache which is not being updated. You can find the old config file locally with 

       
        from deepdoctection.utils import get_configs_dir_path
        tesseract_config_path = os.path.join(get_configs_dir_path(),"dd/conf_tesseract.yaml")

and replace with the new config file 

https://github.com/deepdoctection/deepdoctection/blob/master/configs/dd/conf_tesseract.yaml      
         
I suspect the file in your cache will not have the now needed line:

       LINES: False

All configs downloaded from the Huggingface hub will be downloaded again once changed. However, the Tesseract config file will be copied from the source code into your cache only when you instantiate an analyzer for the first time. 

Hope this helps !

**Comment by crypt3dC4rb0n** [comment]
Thanks for the quick response. I've applied the fix you suggested and it is working fine now. 👍
